### DISCUSSION TITLE
Connect to jupyter server

### DISCUSSION THREAD

**Discussion by ando600** [blinksh/blink#1702] [Support]
Hi 
I am just getting started with blink, and it seems like a great tool. I really want to use it to work on my jupyter notebooks for a remote data science workflow. I know about tools like juno, but I want to use blink, as it is the only one that allows vim bindings to work properly, and I need vim to effectively navigate, especially on iPad. I have tried to connect the the blink vscode instance to a remote jupyter server, as well as connecting blink to a code-server I have setup on my larger compute server, and neither work for some reason. 

As for connecting to the remote jupyter server, I get the error:
"when connecting from vscode.dev jupyter servers must be started with specific options to connect"

And when I try to connect to the code-server instance, (where I would then connect to the local jupyter server), 
I get issues since my code server has an IP available through tailscale, and is not a github server. I have tried port forwarding, to get the vscode server to show up on a local port, but when I try to visit the code-server instance in the browser, I just get stuck at a loading screen, and blink cannot connect to the locahost either. 

Please advise. I just want remote jupyter lab on my iPad with vim bindings. However I get it, I don't care.

Thanks,
Andrew

**Comment by ando600** [comment]
As an an addendum, for the error "when connecting from vscode.dev jupyter servers must be started with specific options to connect", I have tried adding the --NotebookApp.allow_orign="*" flag on the jupyter lab/notebook command, and it did not help.

**Comment by carloscabanero** [comment]
Hey Ando, thanks for the message. I have tried connecting to code-server and Jupyter using the new Blink Build and MS VS code-server (available for insiders), and that works really well.

What code-server are you using? I may try to replicate this next week.

**Reply by ando600** [reply]
Thanks for the reply. I am using this one: 
https://github.com/coder/code-server

If I should switch servers let me know, I can set up a different one.

**Reply by carloscabanero** [reply]
So for Blink Build I played with VS Code Server and I think it is probably the easiest to use https://code.visualstudio.com/blogs/2022/07/07/vscode-server.

It comes as a binary, you can install it in your machine or in Blink Build and then connecting to it with Blink Code is straightforward. They now start the kernels and all the Jupyter machinery themselves so you don't need to connect to a running instance yourself too.

Give it a try and let me know how it goes.
